### PR TITLE
Mk/julification

### PR DESCRIPTION
**Pull request by kalmarek**

### CODE REVIEW

**Comment by kalmarek** [comment]
@piotrmizerka you need to give me write access

File: src/certification.jl [hunk 1]
```diff
@@ -1,52 +1,30 @@
...
     # Changing Q to the corresponding interval-entry matrix
-    Q_interval = reshape([@interval(0) for i in 1:(mn*mn)], mn, mn)
-    for i in 1:mn
-        for j in i:mn
-            Q_interval[i,j] = @interval(Q[i,j])
-            Q_interval[j,i] = Q_interval[i,j]
-        end
-    end
+    Q_interval = map(x->@interval(x), Symmetric((Q.+Q')./2))
+    P_interval_RG = RG.(Q_interval^2)
```
Comment (piotrmizerka) [inline src/certification.jl:8]: Transpose will do as well

File: src/positive_approx.jl [hunk 2]
```diff
@@ -59,22 +59,19 @@ function sos_problem_matrix(M, order_unit, upper_bound::Float64=Inf)
...
 
-function sos_problem_solution_scs(sos_problem, is_silent = false)
-   with_scs = with_optimizer(SCS.Optimizer, eps=1e-5, acceleration_lookback=0, max_iters = 5000000) # one may try to enlarge acceleration_lookback
-   set_optimizer(sos_problem, with_scs)
-   if is_silent
-      set_silent(sos_problem)
-   end
-   optimize!(sos_problem)
-   λ, P = value(sos_problem[:λ]), value.(sos_problem[:P])
+function sos_problem_solution(sos_problem; optimizer)
```
Comment (piotrmizerka) [inline src/positive_approx.jl:84]: ; allows to call function like this ...(optimizer = ...)

File: src/positive_approx.jl [hunk 3]
```diff
@@ -59,22 +59,19 @@ function sos_problem_matrix(M, order_unit, upper_bound::Float64=Inf)
...
+   JuMP.set_optimizer(sos_problem, optimizer)
+   JuMP.optimize!(sos_problem)
+   λ, P = JuMP.value(sos_problem[:λ]), JuMP.value.(sos_problem[:P])
 
    return λ, P
```
Comment (piotrmizerka) [inline src/positive_approx.jl]: Maybe it will be good to return termination status as well

File: src/fox_derivatives.jl [hunk 4]
```diff
@@ -46,20 +46,19 @@ end
...
+    sizehint!(half_basis, length(relations)*maximum(length∘word, relations))
 
-    function relation_append_basis(u::FPGroupElement)
-        for k in 1:length(word(u))
-            for l in k:length(word(u))
-                append!(half_basis, [F(word(u)[k:l])])
+    for rel in relations
+        for k in 1:length(word(rel))
+            for l in k:length(word(rel))
+                append!(half_basis, [F(word(rel)[k:l])])
```
Comment (kalmarek) [inline src/fox_derivatives.jl:59]: could be replaced by
```julia
function all_subwords(w::AbstractWord)
...
end

for rel in relations
    append!(half_basis, [F(sw) for sw in all_subwords(word(rel))])
end
```

File: test/runtests.jl [hunk 5]
```diff
@@ -2,13 +2,40 @@ using StarAlgebras
...
+end
+
 @testset "LowCohomologySOS" begin
    include("group_rings_tests.jl")
    include("fox_derivatives_tests.jl")
    include("positive_approx_tests.jl")
    include("certification_tests.jl")
-end
\ No newline at end of file
+end
```
Comment (piotrmizerka) [inline test/runtests.jl:41]: Function scs_opt better be put in "positive_approx_tests.jl"

**Review by piotrmizerka — APPROVED** [review]

**Review by piotrmizerka — APPROVED** [review]

**Review by piotrmizerka — COMMENTED** [review]
I prefer multiple @infos as the output seems more readible to me.

**Pull request merged**